en/docs/developer-portal/manage-application.md

```diff
@@ -8,46 +8,150 @@ Let's get started!
 
 This section will walk you through the steps to generating an API access token in Choreo. 
 
 Let's get started!
 
-1. Sign in to Choreo Developer Portal at [https://devportal.choreo.dev](https://devportal.choreo.dev). Alternatively, click the Developer Portal link on the Choreo console header. 
+### Access Token for Production 
 
-    ![Developer Portal](../assets/img/developer-portal/manage-applications/developer-portal.png){.cInlineImage-half}
+Depending on your use case, you can generate an access token using one of the two grant types available in Choreo: Client Credentials and Token Exchange. 
+
+!!! note
+    We recommend you use the **Client Credentials** grant type to obtain an access token **for testing purposes** and use the **Token Exchange** grant type **for production use**. 
+
+Token Exchange grant type requires you to pass a subject_token as a parameter in the token invocation call. You can obtain a subject token (id token) by using the `open_id` scope. Follow the steps below to generate an access token from the IdP along with the subject token:
 
+1. Register the IdP in Choreo by following the steps in the section [add an external IdP](../../manage/connect-to-an-external-identity-provider/#step-1-add-an-external-idp/).
+
+2. Generate a token from the registered IdP with the `openId` scope and retrieve the `id_token` from the token response.  
+                
+    ``` json tab="Format"
+        {"access_token":"<access__token>","refresh_token":"<refresh_token>","scope":"openid","id_token":"<id_token>","token_type":"Bearer","expires_in":3600}
+    ```
+
+3. Sign in to Choreo Developer Portal at [https://devportal.choreo.dev](https://devportal.choreo.dev). Alternatively, click the Developer Portal link on the Choreo console header. 
+4. Click **Applications**.
+5. Click **Production Keys** and **OAuth 2.0 Tokens** on the left pane.
+6. Expand the **Advanced Configurations** section and review the options. 
+
+    ![Advanced Configurations](../assets/img/developer-portal/manage-applications/advanced-configurations.png){.cInlineImage-half}
+
+    - **Grant Types:** The grant types used to generate the access token. Keep Token Exchange and Refresh Token grant type selected.
+    - **Public Client:** Identify the application as a public client to allow authentication without a client secret. You can use this for applications running on a browser or mobile device. 
+    - **Application access token expiry time:** The access token expiry time (seconds).
+    - **Refresh token expiry time:** The refresh token expiry time (seconds).
+    - **ID token expiry time:** ID token expiry time (seconds).
+
+7. Click **Generate Credentials** to generate the credentials for the application for the first time. Copy the **Consumer Key** value.
+8. Generate an access token by invoking the token endpoint using the Token Exchange grant type as follows: 
```

Comment:
<application_consumer_key>

Reply:
done

Reply:
This is still missing a `<`

Reply:
Yes, you're right. The changes wern't pushed. Added them now.